src/plugins/intel_npu/tests/functional/behavior/npuw/behavior_tests.cpp

```diff
@@ -53,209 +61,470 @@ TEST_F(BehaviorTestsNPUW, TestInfrastructureIsCorrect) {
+    auto compiled_model = core.compile_model(model, "NPU", use_npuw_props);
+    EXPECT_ANY_THROW(compiled_model.create_infer_request());
+}
+
+TEST_F(BehaviorTestsNPUWOnlinePartitioning, InferIsSuccessful) {
+    model = model_generator.get_model_with_repeated_blocks();
+
+    // Set expectations first:
+    EXPECT_COMPILE_MODEL(npu, 12);
+    EXPECT_COMPILE_MODEL(cpu, 0);
+
+    for (int i = 0;  i < 12; i++) {
+        EXPECT_CREATE_SYNC_INFER_REQ(npu, MODEL(i), 1);
+        EXPECT_INFER(npu, MODEL(i), 1);
+    }
+
+    // Register mock objects as plugins in OpenVINO:
+    register_mock_plugins_in_ov();
+
+    // Do the actual test:
+    use_npuw_props.emplace(devices("MockNPU"));
+    use_npuw_props.emplace(partitioning::online::min_size(12));
+    use_npuw_props.emplace(partitioning::online::keep_blocks(9));
+    auto compiled_model = core.compile_model(model, "NPU", use_npuw_props);
+    auto infer_request = compiled_model.create_infer_request();
+    EXPECT_NO_THROW(infer_request.infer());
+}
+
+TEST_F(BehaviorTestsNPUWOnlinePartitioning, InferIsFailSafe) {
+    model = model_generator.get_model_with_repeated_blocks();
+
     // Set expectations first:
     {
         InSequence seq;
-        EXPECT_CALL(*npu_plugin, compile_model(A<const std::shared_ptr<const ov::Model>&>(), _)).Times(1);
-        EXPECT_CALL(*cpu_plugin, compile_model(A<const std::shared_ptr<const ov::Model>&>(), _)).Times(0);
+        EXPECT_COMPILE_MODEL(npu, 12);
+        EXPECT_COMPILE_MODEL(cpu, 1);
     }
-    npu_plugin->set_expectations_to_comp_models(0, [](MockCompiledModel& model) {
-        EXPECT_CALL(model, create_sync_infer_request()).Times(1);
-    });
-    npu_plugin->set_expectations_to_infer_reqs(0, [](MockInferRequest& request) {
-        EXPECT_CALL(request, infer())
-            .Times(1)
-            .WillOnce(Throw(std::runtime_error("Infer on MockNPU is failed")));
-    });
+    for (int i = 0;  i < 12; i++) {
+       EXPECT_CREATE_SYNC_INFER_REQ(npu, MODEL(i), 1);
+    }
+    for (int i = 0;  i < 11; i++) {
+        EXPECT_INFER(npu, MODEL(i), 1);
+    }
+    EXPECT_INFER(npu, MODEL(11), 1, THROW("Infer on MockNPU is failed"));
+
+    EXPECT_CREATE_SYNC_INFER_REQ(cpu, MODEL(0), 1);
+    EXPECT_INFER(cpu, MODEL(0), 1);
 
     // Register mock objects as plugins in OpenVINO:
     register_mock_plugins_in_ov();
 
     // Do the actual test:
-    auto npuw_properties = 
-        ov::AnyMap{ov::intel_npu::use_npuw(true),
-                   ov::intel_npu::npuw::devices("MockNPU")};
-    auto compiled_model = core.compile_model(model, "NPU", npuw_properties);
+    use_npuw_props.emplace(devices("MockNPU,MockCPU"));
+    use_npuw_props.emplace(partitioning::online::min_size(12));
+    use_npuw_props.emplace(partitioning::online::keep_blocks(9));
+    auto compiled_model = core.compile_model(model, "NPU", use_npuw_props);
+    auto infer_request = compiled_model.create_infer_request();
+    EXPECT_NO_THROW(infer_request.infer());
```

Comment:
Seems like there is some copied code. Could you move it into a macro or a function?

Reply:
Do you think we need this for actions?